google/cloud/sqlalchemy_spanner/sqlalchemy_spanner.py

```diff
@@ -798,13 +798,14 @@ def do_rollback(self, dbapi_connection):
         To prevent rollback exception, don't rollback
         committed/rolled back transactions.
         """
-        if (
-            not isinstance(dbapi_connection, spanner_dbapi.Connection)
-            and dbapi_connection.connection._transaction
-            and (
-                dbapi_connection.connection._transaction.rolled_back
-                or dbapi_connection.connection._transaction.committed
-            )
+        if isinstance(dbapi_connection, spanner_dbapi.Connection):
+            dbapi_connection = dbapi_connection
```

Comment:
nit: this line is redundant which means this check can be refactored. See:

```
if not isinstance(dbapi_connection, spanner_dbapi.Connection):
            dbapi_connection = dbapi_connection.connection
```

Reply:
Done